MCprep_addon/spawner/models.py

```diff
@@ -0,0 +1,1350 @@
+				# edge block rotation 3
+				new_ob.rotation_euler[2]+= math.pi/2
+			elif rot==8:
+				# edge block rotation 4 (ceiling, not 'keep same')
+				new_ob.rotation_euler[0]+= math.pi/2
+
+			# extra variance to break up regularity, e.g. for tall grass
+			# first, xy and z variance
+			if [True,1] == swapProps['variance']:
+				x = (random.random()-0.5)*0.5
+				y = (random.random()-0.5)*0.5
+				z = (random.random()/2-0.5)*0.6
+				new_ob.location += mathutils.Vector((x, y, z))
+			# now for just xy variance, base stays the same
+			elif [True,0] == swapProps['variance']: # for non-z variance
+				x = (random.random()-0.5)*0.5    # values LOWER than *1.0 make it less variable
+				y = (random.random()-0.5)*0.5
+				new_ob.location += mathutils.Vector((x, y, 0))
+
+			# Clear selection before moving on with next iteration
+			for ob in context.selected_objects:
+				if util.bv28():
+					continue
+				util.select_set(ob, False)
+		return grouped, dupedObj
+
+
+	def offsetByHalf(self, obj):
+		if obj.type != 'MESH':
+			return
+		conf.log("doing offset")
+		active = bpy.context.object #preserve current active
+		util.set_active_object(bpy.context, obj)
+		bpy.ops.object.mode_set(mode='EDIT')
+		bpy.ops.mesh.select_all(action='SELECT')
+		bpy.ops.transform.translate(value=(0.5, 0.5, 0.5))
+		bpy.ops.object.mode_set(mode='OBJECT')
+		obj.location[0] -= .5
+		obj.location[1] -= .5
+		obj.location[2] -= .5
+		util.set_active_object(bpy.context, active)
+
+
+class MCPREP_OT_fix_mineways_scale(bpy.types.Operator):
+	"""Quick upscaling of Mineways import by 10 for modelping"""
+	bl_idname = "object.fixmodelsize"
+	bl_label = "Mineways quick upscale"
+	bl_options = {'REGISTER', 'UNDO'}
+
+	@tracking.report_error
+	def execute(self, context):
+		conf.log("Attempting to fix Mineways scaling for model")
+
+		# get cursor loc first? shouldn't matter which mode/location though
+		tmp_loc = util.get_cuser_location(context)
+		if hasattr(context.space_data, "pivot_point"):
+			tmp = context.space_data.pivot_point
+			bpy.context.space_data.pivot_point = 'CURSOR'
+		else:
+			tmp = context.scene.tool_settings.transform_pivot_point
+			context.scene.tool_settings.transform_pivot_point = 'CURSOR'
+		util.set_cuser_location((0,0,0), context)
+
+		bpy.ops.transform.resize(value=(10, 10, 10))
+		bpy.ops.object.transform_apply(location=False, rotation=False, scale=True)
+
+		if hasattr(context.space_data, "pivot_point"):
+			bpy.context.space_data.pivot_point = tmp
+		else:
+			context.scene.tool_settings.transform_pivot_point = tmp
+		util.set_cuser_location(tmp_loc, context)
+		return {'FINISHED'}
+
+
+# -----------------------------------------------------------------------------
+#       Register functions
+# -----------------------------------------------------------------------------
+
+
+classes = (
+	MCPREP_OT_model_path_reset,
+	MCPREP_OT_model_spawner,
+	MCPREP_OT_reload_models,
+	MCPREP_UL_model_list,
+	MCPREP_OT_model,
+	MCPREP_OT_fix_mineways_scale,
```

Comment:
Remember to remove the operators here that are no longer needed (e.g. MCPREP_OT_fix_mineways_scale and MCPREP_OT_model)

Reply:
I was planning to use it to auto replace minecarts and such things. See a previous comment I made